e2e-tests/src/test/button_game/mod.rs

```diff
@@ -7,29 +7,34 @@ use helpers::sign;
 use log::info;
 
 use crate::{
-    test::button_game::helpers::{
-        assert_recv, assert_recv_id, refute_recv_id, setup_button_test, wait_for_death,
-        ButtonTestContext,
+    test::button_game::{
+        contracts::MarketplaceInstance,
+        helpers::{
+            assert_recv, assert_recv_id, refute_recv_id, setup_button_test, wait_for_death,
+            ButtonTestContext,
+        },
     },
     Config,
 };
 
 mod contracts;
 mod helpers;
 
-/// Tests trading on the marketplace.
+/// Tests trading on the marketplace (with update).
 ///
 /// The scenario:
 ///
 /// 1. Buys a ticket without setting the max price (this should succeed).
 /// 2. Tries to buy a ticket with setting the max price too low (this should fail).
 /// 3. Tries to buy a ticket with setting the max price appropriately (this should succeed).
-pub fn marketplace(config: &Config) -> Result<()> {
+///
+/// Additionally there is an update performed amidst contract's operation.
```

Comment:
Suggestion: if you decide to keep the structure of the test you're proposing, then update the comment with the actual sequence of steps instead of a mysterious "there is an update performed at an unknown point".